server/repository/src/db_diesel/temperature_chart/temperature_chart_row.rs

```diff
@@ -0,0 +1,224 @@
+use crate::{db_diesel::temperature_log_row::temperature_log, DBType};
+use chrono::NaiveDateTime;
+use diesel::{
+    backend::Backend, expression::*, prelude::*, query_builder::*, query_source::*, sql_types::*,
+};
+
+#[derive(Debug, PartialEq)]
+pub struct TemperatureChartRow {
+    pub from_datetime: NaiveDateTime,
+    pub to_datetime: NaiveDateTime,
+    pub average_temperature: f64,
+    pub sensor_id: String,
+}
+
+// Needed for allow_table_to_appear_in_same_query
+
+pub use TemperatureChart as table;
+allow_tables_to_appear_in_same_query!(temperature_log, self);
+
+#[derive(Debug, Clone, QueryId)]
+pub struct TemperatureChart {
+    pub(super) intervals: Vec<Interval>,
+}
+#[derive(Debug, Clone, PartialEq)]
+pub struct Interval {
+    pub from_datetime: NaiveDateTime,
+    pub to_datetime: NaiveDateTime,
+}
+
+impl QueryFragment<DBType> for TemperatureChart {
```

Comment:
Below should be pretty straight forward if you look at explanation in README.md

Reply:
I think it might be good to add a mention of the README.md in a comment here, that way you'll encourgage people to find and read it if they come across this code in the future.

Reply:
Agree, thanks 97fabf11028a98d46eba6cae12df750dfc5b606e